server/protocol.go

```diff
@@ -104,134 +52,164 @@ type Bytestring struct {
 	Bytes []byte
 }
 
-func parseAttestEK(ek *attest.EK) (RegistrationData, error) {
+func parseAttestEK(ek *attest.EK) (EnrollData, error) {
 	if reflect.TypeOf(ek.Public).String() != "*rsa.PublicKey" {
-		return RegistrationData{}, errors.New("Invalid key type:" + reflect.TypeOf(ek.Public).String())
+		return EnrollData{}, errors.New("Invalid key type:" + reflect.TypeOf(ek.Public).String())
 	}
 	var c []byte = make([]byte, 0)
 	if ek.Certificate != nil {
 		c = ek.Certificate.Raw
 	}
 	var n = ek.Public.(*rsa.PublicKey).N.Bytes()
 	var e = ek.Public.(*rsa.PublicKey).E
-	return RegistrationData{c, n, e, *pcrextend}, nil
+	return EnrollData{c, n, e, *pcrextend}, nil
+}
+
+func establishEncryptedSession(ch chan []byte) (*Session, error) {
+	var data Bytestring
+	var key []byte
+	var session = NewSession(ch)
+	var err error
+
+	logrus.Info("Getting client UUID")
+	if err = recvMsg(&data, session); err != nil {
+		return nil, err
+	}
+	if session.uuid, err = uuid.FromBytes(data.Bytes); err != nil {
+		close(ch)
+		return nil, err
+	}
+
+	if *enroll {
+		// TODO: Save key somewhere
+	} else {
+		// TODO: Fetch key from persistence point
+	}
+	if err != nil {
+		close(ch)
+		return nil, err
+	}
+	if session.Encrypt(key); err != nil {
```

Comment:
There is a bug here, it should be err = session.Encrypt(key) I guess?
Doesn't the go compiler warn about ignoring an error return type?

Reply:
Yes, this is obviously a bug, and no, the go compiler doesn't complain at all.